src/home_robot/home_robot/motion/stretch.py

```diff
@@ -660,14 +673,27 @@ def manip_ik(
             # So how do we do that?
             # This logic currently in local hello robot client
             raise NotImplementedError()
-        _q = self.manip_ik_solver.compute_ik(
-            pos, quat, q0, num_attempts=num_attempts, verbose=verbose
-        )
-        if _q is None:
-            return None
-        if self._ik_type == "pinocchio":
+
+        # TODO: simplify the following by performing ik-type specific conversions out of this method
+        if self._ik_type == "pybullet":
+            _q = self.manip_ik_solver.compute_ik(
+                pos,
+                quat,
+                default_q,
+                num_attempts=num_attempts,
+                verbose=verbose,
+            )
+            if _q is None:
+                return None
+            q = self._from_manip_format(_q, q0)
+        elif self._ik_type == "pinocchio":
+            _q = self.manip_ik_solver.compute_ik(
+                pos, quat, self._ros_pose_to_pinocchio(q0)
+            )
+            if _q is None:
+                return None
             _q = _q[0]
-        q = self._from_manip_format(_q, default_q)
+            q = self._pinocchio_pose_to_ros(_q)
```

Comment:
Add more comments as to what exactly the two `compute_ik` returns? Would make this part easier to change in the future.

Reply:
Actually maybe we should wait for #121 to be merged, seems like it deals with a lot of issues here.